.travis.yml

```diff
@@ -14,18 +14,37 @@ matrix:
       os: osx
       osx_image: beta-xcode6.2
       rvm: system
+    - env: RUBY=1.8.7
+      os: osx
+      osx_image: xcode7
+      rvm: 1.8.7
+      language: ruby
+    - env: LINUX=1
+      os: linux
+      rvm: 1.8.7
+      sudo: false
+      language: ruby
 
 before_install:
   - export TRAVIS_COMMIT=$(git rev-parse --verify -q HEAD)
-  - cd /usr/local
-  - sudo rm -rf /usr/local/.git/refs /usr/local/.git/packed-refs
-  - sudo rsync -az "$TRAVIS_BUILD_DIR/" /usr/local/
-  - export TRAVIS_BUILD_DIR="/usr/local"
-  - sudo chown -R $USER /usr/local
-  - env | grep TRAVIS_
-  - if [ -f ".git/shallow" ]; then travis_retry git fetch --unshallow; fi
+  - if [ -n "$OSX" ]; then
+      cd /usr/local;
+      sudo rm -rf /usr/local/.git/refs /usr/local/.git/packed-refs;
+      sudo rsync -az "$TRAVIS_BUILD_DIR/" /usr/local/;
+      export TRAVIS_BUILD_DIR="/usr/local";
+      sudo chown -R $USER /usr/local;
+    elif [ -n "$RUBY" ]; then
+      export HOMEBREW_OLD_RUBY_TEST="1";
+    elif [ -n "$LINUX" ]; then
+      export HOMEBREW_OLD_RUBY_TEST="1";
+      export HOMEBREW_EMULATE_MAC="1";
```

Comment:
If possible, I'd prefer to set env using travis `env` DSL. The main reason is if we keep each command as a entry, Travis will fold them nicely one by one(instead of one big block).